src/explore-education-statistics-common/src/utils/url/applyRedirectRules.ts

```diff
@@ -0,0 +1,93 @@
+import redirectService, {
+  RedirectType,
+  redirectPathStarts,
+} from '@common/services/redirectService';
+
+const prodPublicUrl = 'https://explore-education-statistics.service.gov.uk/';
```

Comment:
Is it the right approach to use this as the base? If we're on non-prod environments, won't this potentially result in incorrect links in those environments?

I mentioned this in your other PR, but we should probably set the host's URL using `setHostUrl` / `getHostUrl` functions and use this instead?